### DISCUSSION TITLE
Convert date from Date picker format to the one accepted by my REST API

### DISCUSSION THREAD

**Discussion by dominikkubik** [Budibase/budibase#12620] [Help] [answered]
Hi

I would like to ask for help.
Using the Form and Date picker I am getting the date as: 2023-12-17T00:00:00.000 which is then using Bindings forwarded to one POST REST API Endpoint for external system.
But i am getting error [ "The purchase date must be a valid date in YYYY-MM-DD format"]
Is there any possibility how to convert the date into form accepted by my external system?
I tried all possible combination i knew, even some Javascript hints i found here, but it is still not working.

Any idea?
Thank you

**Comment by ConorWebb96** [comment, marked as answer]
Hey @dominikkubik,

This isn't really a Budibase issue, from the looks of the error you are getting is from the API you are posting to. You need to convert Budibase's date fields value to `yyyy-mm-dd`. This code below should do this within Budibase.

```js
const dateString = "2023-12-17T00:00:00.000"; // replace with your field binding
const date = new Date(dateString);
const year = date.getFullYear();
const month = String(date.getMonth() + 1).padStart(2, '0');
const day = String(date.getDate()).padStart(2, '0');
const formattedDate = `${year}-${month}-${day}`;

return formattedDate;
```

I'd recommend converting it to state then binding the state value to where you are posting the data to. 

Hopefully this helps.

**Reply by dominikkubik** [reply]
Thank you for quick reply.
I forget to mention that "But i am getting error [ "The purchase date must be a valid date in YYYY-MM-DD format"]" is error from API from another system, of course not issue in Budibase.

My answer was regarding the possibility of formatting date in Budibase.
Thank you for your answer.
